libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -96,6 +96,18 @@ void AP_AHRS_NavEKF::update(bool skip_ins_update)
         update_EKF3();
         update_EKF2();
     }
+
+    // update home location for GPS baro correction for DCM
+    Location EKF_home_loc;
+    if (_ekf_type == 2 && _ekf2_started && (EKF2.getOriginHgtMode() & (1<<2)) && EKF2.healthy()) {
+        EKF2.getOriginLLH(-1, EKF_home_loc);
```

Comment:
The vertical position of the EKF origin will only drift if the EK2/3_OGN_MASK parameter is set with bit position 2 true. However the horizontal position should be stable once GPS checks have passed.